scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -155,7 +155,27 @@
   [dru]
   (* dru metrics-dru-scale))
 
-(defn compute-pending-job-dru
+(defn compute-pending-gpu-job-dru
+  [{:keys [task->scored-task user->sorted-running-task-ents user->dru-divisors] :as state}
+   pending-job-ent]
+  (let [user (:job/user pending-job-ent)
+        {gpu-req :gpus} (util/job-ent->resources pending-job-ent)
+        gpu-divisor (get-in user->dru-divisors [user :gpus])
+        pending-task-ent (util/create-task-ent pending-job-ent)
+        nearest-task-ent (some-> user->sorted-running-task-ents
+                                 (get user)
+                                 (rsubseq <= pending-task-ent)
+                                 (first))
+        nearest-task-dru (if nearest-task-ent
+                           (get-in task->scored-task [nearest-task-ent])
+                           0.0)
+        pending-job-dru (+ nearest-task-dru (/ gpu-req gpu-divisor))]
+    (histograms/update! pending-job-drus (dru-at-scale pending-job-dru))
+    (histograms/update! nearest-task-drus (dru-at-scale nearest-task-dru))
+
+    pending-job-dru))
+
+(defn compute-pending-normal-job-dru
```

Comment:
normal as opposed to gpu? That seems odd since you have a case for gpu in the function.